server/bundler/assets-writer.js

```diff
@@ -0,0 +1,101 @@
+/** @format */
+/**
+ * External Dependencies
+ */
+const fs = require( 'fs' ); // eslint-disable-line  import/no-nodejs-modules
+const path = require( 'path' );
+
+function AssetsWriter( options ) {
+	this.options = Object.assign(
+		{
+			path: './build',
+			filename: 'assets.json',
+			stats: {
+				hash: true,
+				publicPath: true,
+				assets: true,
+				children: false,
+				chunks: true,
+				chunkModules: false,
+				chunkOrigins: false,
+				entrypoints: true,
+				modules: false,
+				source: false,
+				errorDetails: true,
+				timings: false,
+				reasons: false,
+			},
+		},
+		options
+	);
+	this.createOutputStream();
+}
+
+Object.assign( AssetsWriter.prototype, {
+	createOutputStream: function() {
+		this.outputPath = path.join( this.options.path, this.options.filename );
+		this.outputStream = fs.createWriteStream( this.outputPath );
+	},
+	apply: function( compiler ) {
+		const self = this;
+
+		compiler.hooks.afterEmit.tap( 'AssetsWriter', compilation => {
+			const stats = compilation.getStats().toJson( this.options.stats );
+
+			const statsToOutput = {};
+			statsToOutput.publicPath = stats.publicPath;
+			statsToOutput.manifests = {};
+
+			for ( const name in stats.assetsByChunkName ) {
+				// make the manifest inlineable
+				if ( String( name ).startsWith( 'manifest' ) ) {
+					statsToOutput.manifests[ name ] = compilation.assets[
+						stats.assetsByChunkName[ name ]
+					].source();
+				}
+			}
+
+			statsToOutput.entrypoints = stats.entrypoints;
```

Comment:
maybe we could set it to the right value the first time with some help from lodash:

```js
statsToOutput.entrypoints = mapValues( stats.entrypoints, entry => ( {
  chunks: reject( entry.assets, 'manifest' ),
  assets: reject( entry.assets, 'manifest' )
} ) );
```

Reply:
done